CondCore/PhysicsToolsPlugins/plugins/PfCalibration_PayloadInspector.cc

```diff
@@ -0,0 +1,132 @@
+#include "CondFormats/DataRecord/interface/PFCalibrationRcd.h"
+
+#include <memory>
+#include <sstream>
+#include <fstream>
+#include <iostream>
+#include <array>
+#include <map>
+
+// include ROOT
+#include "TH2F.h"
+#include "TF1.h"
+#include "TLegend.h"
+#include "TCanvas.h"
+#include "TLine.h"
+#include "TStyle.h"
+#include "TLatex.h"
+#include "TPave.h"
+#include "TPaveStats.h"
+
+using namespace cond::payloadInspector;
+
+class PerformancePayloadFromTFormulaExposed : public PerformancePayloadFromTFormula {
+public:
+  int resultPos(PerformanceResult::ResultType rt) const override {
+    return PerformancePayloadFromTFormula::resultPos(rt);
+  }
+};
+
+static std::map<PerformanceResult::ResultType, std::string> functType = {
+    {PerformanceResult::PFfa_BARREL, "PFfa_BARREL"},
+    {PerformanceResult::PFfa_ENDCAP, "PFfa_ENDCAP"},
+    {PerformanceResult::PFfb_BARREL, "PFfb_BARREL"},
+    {PerformanceResult::PFfb_ENDCAP, "PFfb_ENDCAP"},
+    {PerformanceResult::PFfc_BARREL, "PFfc_BARREL"},
+    {PerformanceResult::PFfc_ENDCAP, "PFfc_ENDCAP"},
+
+    {PerformanceResult::PFfaEta_BARRELH, "PFfaEta_BARRELH"},
+    {PerformanceResult::PFfaEta_ENDCAPH, "PFfaEta_ENDCAPH"},
+    {PerformanceResult::PFfbEta_BARRELH, "PFfbEta_BARRELH"},
+    {PerformanceResult::PFfbEta_ENDCAPH, "PFfbEta_ENDCAPH"},
+    {PerformanceResult::PFfaEta_BARRELEH, "PFfaEta_BARRELEH"},
+    {PerformanceResult::PFfaEta_ENDCAPEH, "PFfaEta_ENDCAPEH"},
+    {PerformanceResult::PFfbEta_BARRELEH, "PFfbEta_BARRELEH"},
+    {PerformanceResult::PFfbEta_ENDCAPEH, "PFfbEta_ENDCAPEH"},
+
+    {PerformanceResult::PFfaEta_BARREL, "PFfaEta_BARREL"},
+    {PerformanceResult::PFfaEta_ENDCAP, "PFfaEta_ENDCAP"},
+    {PerformanceResult::PFfbEta_BARREL, "PFfbEta_BARREL"},
+    {PerformanceResult::PFfbEta_ENDCAP, "PFfbEta_ENDCAP"},
+
+    {PerformanceResult::PFfcEta_BARRELH, "PFfcEta_BARRELH"},
+    {PerformanceResult::PFfcEta_ENDCAPH, "PFfcEta_ENDCAPH"},
+    {PerformanceResult::PFfdEta_ENDCAPH, "PFfdEta_ENDCAPH"},
+    {PerformanceResult::PFfcEta_BARRELEH, "PFfcEta_BARRELEH"},
+    {PerformanceResult::PFfcEta_ENDCAPEH, "PFfcEta_ENDCAPEH"},
+    {PerformanceResult::PFfdEta_ENDCAPEH, "PFfdEta_ENDCAPEH"}};
+
+template <PerformanceResult::ResultType T>
+class PfCalibration : public cond::payloadInspector::PlotImage<PerformancePayloadFromTFormula, SINGLE_IOV> {
+public:
+  PfCalibration()
+      : cond::payloadInspector::PlotImage<PerformancePayloadFromTFormula, SINGLE_IOV>("Performance Payload formula") {}
+  bool fill() override {
+    auto tag = PlotBase::getTag<0>();
+    auto iov = tag.iovs.front();
+    std::string tagname = tag.name;
+    auto payload = fetchPayload(std::get<1>(iov));
+
+    if (!payload.get())
+      return false;
+
+    int pos = ((PerformancePayloadFromTFormulaExposed*)payload.get())->resultPos(T);
+    auto formula = payload->formulaPayload();
+    auto formula_vec = formula.formulas();
+    auto limits_vec = formula.limits();
+
```

Comment:
```suggestion
 if (pos < 0 || pos > (int)formula_vec.size()) {
      edm::LogError("PfCalibartion") << "Will not display image for " << functType[T]
                                     << " as it's not contained in the payload!";
      return false;
    }
```

I gather that `pos` can be `-1` in case the `PerformanceResult::ResultType` formula is not stored in the payload, so you'd better exit gracefully here.